File: bug69_skipped_last_element/bug69_skipped_last_element/main1.py
def run():
    data = [0xBBBB * (i + 1) for i in range(10)]
    acc = 0

    # FIX: changed range(len(data) - 1) → range(len(data))
    for i in range(len(data)):
        acc ^= data[i] * (i + 1)
        acc = (acc * 0x9E3779B97F4A7C15) & 0xFFFFFFFFFFFFFFFF
        acc = ((acc << (i % 11 + 1)) | (acc >> (63 - i % 11))) & 0xFFFFFFFFFFFFFFFF

    # FIX: print computed value as hex string
    print("Result:", hex(acc))

run()

File: bug69_skipped_last_element/bug69_skipped_last_element/main1.java
public class main1 {
    public static void main(String[] args) {
        long[] data = new long[10];
        for (int i = 0; i < 10; i++) data[i] = 0xBBBBL * (i + 1);
        
        long acc = 0;
        // FIX: changed i < 9 → i < 10
        for (int i = 0; i < 10; i++) {
            acc ^= data[i] * (i + 1);
            acc = (acc * 0x9E3779B97F4A7C15L);
            acc = ((acc << (i % 11 + 1)) | (acc >>> (63 - (i % 11))));
        }

        // FIX: print computed value as hex string
        System.out.println("Result: " + Long.toHexString(acc));
    }
}